pointcloud/basic_octree_utils.go

```diff
@@ -8,31 +8,40 @@ import (
 )
 
 // Creates a new LeafNodeEmpty.
-func newLeafNodeEmpty() basicOctreeNode {
+func newLeafNodeEmpty(parent *BasicOctree, depth int) basicOctreeNode {
 	octNode := basicOctreeNode{
 		children: nil,
+		parent:   parent,
 		nodeType: leafNodeEmpty,
 		point:    PointAndData{},
+		maxProb:  -1.,
```

Comment:
[q] why is this -1 and not 0?

Reply:
We may in the future want to send points that have 0 probability. For example, if we want to distinguish between areas that have been explored and determined to be empty versus unexplored regions. This topic is still up for debate but is a potential use case. therefore, I think a probability of -1 is a bit more robust

Reply:
Can we make this a constant then?  Something like `const unexplored = -1`